be/src/exprs/json_functions.cpp

```diff
@@ -412,41 +419,47 @@ rapidjson::Value* JsonFunctions::get_json_object_from_parsed_json(
     return root;
 }
 
-void JsonFunctions::json_path_prepare(doris_udf::FunctionContext* context,
+void JsonFunctions::json_function_prepare(doris_udf::FunctionContext* context,
                                       doris_udf::FunctionContext::FunctionStateScope scope) {
     if (scope != FunctionContext::FRAGMENT_LOCAL) {
         return;
     }
 
-    if (!context->is_arg_constant(1)) {
+    if (!context->is_arg_constant(0) && !context->is_arg_constant(1)) {
         return;
     }
+    StringVal* json_str = reinterpret_cast<StringVal*>(context->get_constant_arg(0));
```

Comment:
The first argument may NOT be constant, so `get_constant_arg` may return nullptr.